src/hooks/safe-apps/useSafeApps.ts

```diff
@@ -2,41 +2,84 @@ import * as React from 'react'
 import { SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
 import { useRemoteSafeApps } from '@/hooks/safe-apps/useRemoteSafeApps'
 import { useCustomSafeApps } from '@/hooks/safe-apps/useCustomSafeApps'
+import { usePinnedSafeApps } from '@/hooks/safe-apps/usePinnedSafeApps'
 
 type ReturnType = {
   allSafeApps: SafeAppData[]
+  pinnedSafeApps: SafeAppData[]
+  pinnedSafeAppIds: Set<number>
   remoteSafeApps: SafeAppData[]
   customSafeApps: SafeAppData[]
   remoteSafeAppsLoading: boolean
   customSafeAppsLoading: boolean
   remoteSafeAppsError?: Error
   addCustomApp: (app: SafeAppData) => void
+  togglePin: (appId: number) => void
+}
+
+const useDeadPinnedSafeAppsRemover = (
+  remoteSafeApps: SafeAppData[],
+  pinnedSafeAppIds: Set<number>,
+  updateCallback: (newIds: Set<number>) => void,
+) => {
+  React.useEffect(() => {
```

Comment:
Please import useEffect, useState etc separately, we do it like this in all other components.